### DISCUSSION TITLE
Is it possible to set a custom origin filename and path?

### DISCUSSION THREAD

**Discussion by maiconmva** [spatie/ray#798] [Q&A]
Ray is great but how i can set a custom Origin? 

ex:

function r(...$params) {
    return ray(...$params);
}
This works, except for the fact that Ray shows, ofcourse, the file and line-number of this above ray()-call.

It would be nice if you could configure the depth of the origin-search, so that Ray could show the line where my custom debugging-method was called.

**Comment by howdu** [comment]
Similar request here too.

E.g we log the n+1 violations and need to manually specify the backtrace in the error message.
```php
Model::handleLazyLoadingViolationUsing(static function (Model $model, string $relation): void {
    $fullTrace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, limit: 6);
    $trace = array_pop($fullTrace);

    ray(__(
        'N+1 Query detected for `:relation` on the `:class` model in: :file::line.', [
        'relation' => $relation,
        'class' => $model::class,
        'file' => $trace['file'],
        'line' => $trace['line'],
    ]));
});
```

> N+1 Query detected for `delivery` on the `App\Models\DeliveryItem` model in: www/app/Domain/Logistics/Data/StockAdjustmentData.php:33.
AppServiceProvider.php:61

Instead simple message with the file link at the bottom
> N+1 Query detected for `delivery` on the `App\Models\DeliveryItem` model.
StockAdjustmentData.php:33

**Comment by jmslbam** [comment]
@maiconmva Maybe you can get a bit inspiration from this https://github.com/jmslbam/ray-php-errors
I wrote a custom Payload and mis-used the `label` argument :)

**Reply by jmslbam** [reply]
Oh, sorry, wrong thread
